kyo-prelude/shared/src/main/scala/kyo/Stream.scala

```diff
@@ -328,25 +329,90 @@ final case class Stream[V, -S](v: Ack < (Emit[Chunk[V]] & S)):
+      */
+    def init[V, S](v: => Seq[V] < S, chunkSize: Int = DefaultChunkSize)(using tag: Tag[Emit[Chunk[V]]], frame: Frame): Stream[V, S] =
         Stream[V, S](
-            seq.map { seq =>
+            v.map { seq =>
                 val chunk: Chunk[V] = Chunk.from(seq)
+                val _chunkSize      = chunkSize max 1
                 Emit.andMap(Chunk.empty[V]) { ack =>
                     Loop(chunk, ack) { (c, ack) =>
                         ack match
                             case Stop =>
                                 Loop.done(Stop)
                             case Continue(n) =>
                                 if c.isEmpty then Loop.done(Ack.Continue())
-                                else Emit.andMap(c.take(n))(ack => Loop.continue(c.dropLeft(n), ack))
+                                else
+                                    val i = n min _chunkSize
+                                    Emit.andMap(c.take(i))(ack => Loop.continue(c.dropLeft(i), ack))
                     }
                 }
             }
         )
 
+    /** Creates a stream of integers from start (inclusive) to end (exclusive).
+      *
+      * @param start
+      *   The starting value (inclusive)
+      * @param end
+      *   The ending value (exclusive)
+      * @param step
+      *   The step size (default: 1)
+      * @param chunkSize
+      *   The size of chunks to emit (default: 4096)
+      * @return
+      *   A stream of integers within the specified range
+      */
+    def range[S](start: Int, end: Int, step: Int = 1, chunkSize: Int = DefaultChunkSize)(using
+        tag: Tag[Emit[Chunk[Int]]],
+        frame: Frame
+    ): Stream[Int, S] =
+        if step == 0 || (start < end && step < 0) || (start > end && step > 0) then empty
+        else
+            Stream[Int, S] {
+                val _chunkSize = chunkSize max 1
+
+                Emit.andMap(Chunk.empty[Int]) { ack =>
+                    Loop(start, ack) { (current, ack) =>
+                        ack match
+                            case Stop =>
+                                Loop.done(Stop)
+                            case Continue(n) =>
+                                val continue =
+                                    if step > 0 then current < end
+                                    else current > end
+
+                                if !continue then Loop.done(Stop)
+                                else
+                                    val remaining =
+                                        if step > 0 then
+                                            ((end - current - 1) / step).abs + 1
+                                        else
+                                            ((current - end - 1) / step.abs).abs + 1
+                                    val size  = (n min _chunkSize) min remaining
+                                    val chunk = Chunk.from(Array.range(current, current + size * step, step))
```

Comment:
Could we use `collection.immutable.Range` here? It represents a range without allocating the individual items.